pkg/expr/reader_test.go

```diff
@@ -0,0 +1,143 @@
+package expr
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data/utils/jsoniter"
+	data "github.com/grafana/grafana-plugin-sdk-go/experimental/apis/data/v0alpha1"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReaderReduceMode(t *testing.T) {
+	testData := []struct {
+		name          string
+		bytes         []byte
+		expectedError string
+	}{
+		{
+			name: "no_settings",
+			bytes: []byte(`
+				{
+					"refId": "B",
+					"datasource": {
+						"type": "__expr__",
+						"uid": "__expr__"
+					},
+					"reducer": "last",
+					"expression": "A",
+					"window": "",
+					"type": "reduce"
+				}
+			`),
+			expectedError: "",
```

Comment:
This change here, and everywhere else we see `expectedError: "",`
```suggestion
			expectedError: nil,
```